x-pack/plugins/security/server/lib/authentication/providers/basic.ts

```diff
@@ -39,76 +42,72 @@ export class BasicCredentials {
   }
 }
 
+type RequestWithLoginAttempt = Request & {
+  loginAttempt: () => LoginAttempt;
+};
+
 /**
- * Object that represents available provider options.
- * @typedef {{
- *  protocol: string,
- *  hostname: string,
- *  port: string,
- *  basePath: string,
- *  client: Client,
- *  log: Function
- * }} ProviderOptions
+ * Represents available provider options.
  */
+interface ProviderOptions {
```

Comment:
__note:__ once we typescriptify `Authenticator`, we'll move `ProviderOptions` from all providers to there.